### PR TITLE
Document node exceptions

### PR DESCRIPTION
**Pull request by nh916**
# Description
Wrote first draft of documentation for Node Exceptions. I think this is still a work in progress, but something for us to build on top of.

## Changes
* added type hinting to methods and messages
* formatted error messages
* updated some packages

## Known Issues
*  `CRIPTJsonDeserializationError` docs needs updates down the line
    * more documentation is needed on how to troubleshoot this section
    * better explanation is needed on how and why this error occurs as well as an example of where it would and wouldn't occur
* in order to get the documentation clean with only `Definition` and `How to fix` section, the rest of the method docstrings do not exist
    * I think this is okay as the methods are very simple and do not need much explanation, and simple comments I think might be enough
* I made a bad mistake of creating the sections as `Definition` and `How to Fix` instead of `Definition` and `Troubleshooting` which would have been better, but that can come in future upgrades

## Notes
* Some of these errors are hard to troubleshoot and understand, lets see if we can update and fix them a bit
* We could probably shorten the error names a bit by removing the word `CRIPT` when not needed. 
    * I'm thinking they are using our package so they know that its our errors
    * We could only add `CRIPT` when other packages/modules have the same error name 
         * When I was first creating the `ConnectionError` exception I noticed that 
         [requests](https://requests.readthedocs.io/en/latest/) library has the same exact error name that [requests library had an exception](https://requests.readthedocs.io/projects/cn/zh_CN/latest/_modules/requests/exceptions.html?highlight=connection%20error) named the same thing, 
so I renamed ours to `CRIPTConnectionError` I think in those instances it makes sense, but in other instances we can just remove it

## Screenshot
image

### CODE REVIEW

File: src/cript/api/api.py [hunk 1]
```diff
@@ -436,7 +436,7 @@ def _is_node_schema_valid(self, node_json: str) -> bool:
         try:
             jsonschema.validate(instance=node_dict, schema=db_schema)
         except jsonschema.exceptions.ValidationError as error:
-            raise CRIPTNodeSchemaError(error_message=str(error) + str(error.context)) from error
+            raise CRIPTNodeSchemaError(node_type=node_dict["node"], json_schema_validation_error=str(error))
```
[the remark this answers is not in the document]
Comment (nh916) [reply src/cript/api/api.py:439]: you mean change it back into `error_message`?

[the remark this answers is not in the document]
Comment (InnocentBug) [reply src/cript/api/api.py:439]: no, it should be:
```python
raise CRIPTNodeSchemaError(node_type=node_dict["node"], json_schema_validation_error=str(error)) from error
```

[the remark this answers is not in the document]
Comment (nh916) [reply src/cript/api/api.py:439]: ooh I see what you mean. I'll add that in right now

[the remark this answers is not in the document]
Comment (nh916) [reply src/cript/api/api.py]: done! ✅

**Review by InnocentBug — APPROVED** [review]

**Comment by nh916** [comment]
caught some spelling mistakes with `cspell` and fixed it before merging them in.
that was a great CI tool to add!! very happy with it!

**Pull request merged**